Non-Inclusive Fabric API Compatibility

Leo40Git opened the pull request:
Ignore the kinda iffy name, this PR _should_ allow FAPI compatibility without actually including any FAPI modules ~~and getting the speedrunning community in a tizzy.~~

Currently supported:
- `fabric-rendering-fluids-v1`
- `fabric-rendering-data-attachment-v1`

Leo40Git committed:
create this so i can make a pr

this is just so i don't forget, not actually mergable

MulverineX commented:
There is already a project working on this: https://github.com/comp500/Indium
For now it's a seperate mod that integrates with Sodium, and while it's **not** perfect, it's pretty far along.

jellysquid3 commented:
It's my understanding this PR is in regards to the support Sodium recently gained for some FRAPI modules. Though what exactly it's doing is beyond me.

Leo40Git commented:
this _will_ essentially be a "rewrite" of sorts of the new compat stuff so it won't depend om FAPI beimg included

keyword being will because this is just a placeholder so i don't forget

mrmangohands commented:
For the record we're generally fine with the bundled module approach :P

Leo40Git committed:
do WorldSlice

Leo40Git committed:
do FluidRenderer

Leo40Git committed:
Actually set up CompatHolder

Gradle stuff for testing

Leo40Git commented:
We have actual code now, hooray!

Unfortunately this hasn't been tested yet because `sun.misc.Unsafe` isn't resolving during compilation (what).

Leo40Git committed:
budge fabric_version next to use_fapi

Leo40Git committed:
Fix compilation (Gradle is weird)

Remove hard dependencies on FAPI from FMJ
Remove FabricFluidColorizerAdapter since we can just extend ModelQuadColorProvider<FluidState> in our handler interface

Leo40Git committed:
Improve property name

Leo40Git commented:
Okay, I've tested it - fluid rendering seems to work in both cases.

Leo40Git committed:
Move methods to make more sense context-wise

CompatHolder.createWorldSlice -> WorldSlice.create
CompatHolder.getFluidRenderHandler -> FluidRenderer.getRenderHandler

Leo40Git committed:
uh

Leo40Git committed:
FluidRenderHandlerProvider: Clear cache on resource reload
